### PR TITLE
Update to latest cf-java-client, fix TCK, cmdLineArgs

### PR DESCRIPTION
**Pull request by ericbottard**
Fix AppDeployer to use the full TCK, using the latest spring-cloud-deployer-spi-test-app.

Most importantly, fixes a bug in status where failed states were never witnessed

Shortens app names to 63 chars

Adds support for commandLine params (TCK test skipped, as char escaped are not working yet.
Trivial cases do work though)

Fix unit tests

Add some logging

Fixes spring-cloud/spring-cloud-deployer-cloudfoundry#72
Fixes spring-cloud/spring-cloud-deployer-cloudfoundry#71
Fixes spring-cloud/spring-cloud-deployer-cloudfoundry#69

### CODE REVIEW

File: src/test/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployerTests.java [hunk 1]
```diff
@@ -584,4 +585,9 @@ private String runFullDeployment(AppDeploymentRequest request) throws Interrupte
 		return deployer.deploy(request);
 	}
 
+	private void mockGetUnknownApplication(String applicationName) {
```
Comment (ilayaperumalg) [inline src/test/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployerTests.java]: applicationName may not be required as a param.

File: src/test/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployerIntegrationTests.java [hunk 2]
```diff
@@ -50,75 +42,47 @@
...
-		cloudFoundryAppDeployer = (CloudFoundryAppDeployer) appDeployer;
 	}
 
-	/**
-	 * Doesn't appear like we can enter the failed state, tried for about 3 hrs.
-	 */
 	@Override
-	public void testFailedDeployment() {
-		Assert.isTrue(true);
+	@Ignore("Need to look into args escaping better. Disabling for the time being")
```
Comment (ilayaperumalg) [inline src/test/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployerIntegrationTests.java]: Does this mean, this is a `test` only issue or passing commandline args itself is not working?

Comment (ericbottard) [inline src/test/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployerIntegrationTests.java]: I'd say it's the former, but I haven't had time to look at it yet

**Comment by ilayaperumalg** [comment]
LGTM, merging.

**Pull request merged**